sufia.gemspec

```diff
@@ -31,6 +31,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'select2-rails', '~> 3.5.9'
   spec.add_dependency 'json-schema'
   spec.add_dependency 'oauth'
+  spec.add_dependency 'devise', '~> 4.2'
```

Comment:
No current implementation is forced to do anything.  Sufia 7 is unreleased and still in beta, so the time to modify dependencies is **before release**.  This allows the released version to have a longer lifespan and more consistent profile.  ScholarSphere's tracking *might* be an argument if if Sufia 7 were actually released, or if this PR were targeting a `6.x.x` bugfix, but for now it is just a question of whether consumers want to be tracking the leading edge of development or not.  If a project can't bump deps with a major version release, then it can't do it at all.  

If it were up to me, I would go through and bump up as many dependencies major versions as possible before release.  The release is happening *now* and should use components that are current now.  I.e., get this badge green, or closer to it (especially the oauth gem!): https://gemnasium.com/projecthydra/sufia

Specifically, this would also surface deprecations that we could clean up now rather than allow to creep in with a later `bundle update` or fresh install.